src/calendar/header/index.js

```diff
@@ -65,15 +65,15 @@ class CalendarHeader extends Component {
   onPressLeft() {
     const {onPressArrowLeft} = this.props;
     if(typeof onPressArrowLeft === 'function') {
-      return onPressArrowLeft(this.substractMonth);
+      return onPressArrowLeft(this.substractMonth,this.props.month);
```

Comment:
Why you need the current month returned here? you can have it in 'onVisibleMonthsChange' callback

Reply:
`onVisibleMonthsChange` is not very accurate, it usually gives more than one month.

Reply:
Besides, this isn't a breaking change, so I don't see why including it would be an issue.

Reply:
Also, If I remember correctly, this is **needed** for navigating months in `CalendarList`.

This is because `CalendarList` renders a list of `CalendarListItem` which renders a `Calendar` which includes a `CalendarHeader`, when the left or right navigation arrows of the `CalendarHeader`, instead of navigating along the `CalendarList`, the `Calendar` is updated in place to show the month navigated to.

So for example, if `CalendarList` looked like:

```
-------+-------+-------+-------+
 Index |   0   |   1   |   2   |
-------+-------|-------|-------|
 Month |  Jan  |  Feb  |  Mar  |
-------+-------+-------+-------+
```

When `Jan` is shown, if the right arrow is pressed, instead of moving to the next `CalendarItem` in the `CalendarList` at `1`, The `CalendarItem` at `0` updates itself to show `Feb`. So now, `CalendarList` looks like:

```
-------+-------+-------+-------+
 Index |   0   |   1   |   2   |
-------+-------|-------|-------|
 Month |  Feb  |  Feb  |  Mar  |
-------+-------+-------+-------+
```

This is happening because in `CalendarHeader`:

```
onPressLeft() {
    const {onPressArrowLeft} = this.props;
    if(typeof onPressArrowLeft === 'function') {
        return onPressArrowLeft(this.substractMonth);
    }
    return this.substractMonth();
}

onPressRight() {
    const {onPressArrowRight} = this.props;
    if(typeof onPressArrowRight === 'function') {
        return onPressArrowRight(this.addMonth);
    }
    return this.addMonth();
}
```

which uses:

```
addMonth() {
    this.props.addMonth(1);
}

substractMonth() {
    this.props.addMonth(-1);
}
```

which uses `addMonth` in `Calendar`:

```
addMonth(count) {
    this.updateMonth(this.state.currentMonth.clone().addMonths(count, true));
}
```

which **replaces** the month displayed by `Calendar`.

This behaviour is incorrect when `Calendar` is used **inside** a `CalenderList`. When the left/right arrows of `CalendarHeader`, we need to **move along** the `CalendarList` and display the previous/next `CalendarItem`, not update the month displayed by the active `Calendar` item held by `CalendarList`.

To fix this, I added the following code to `CalendarList` which uses the changes in this MR:

```
onPressArrowLeft={(_,month)=>{
    month = month.clone()
    month.addMonths(-1)
    this.refs.calendarList.scrollToMonth(month)
}}
onPressArrowRight={(_,month)=>{
    month = month.clone()
    month.addMonths(1)
    this.refs.calendarList.scrollToMonth(month)
}}
```

Where 'calendarList` is a reference to the `CalendarList`.

Reply:
To fix this upstream, simply merge this MR, and add the code above for `onPressArrowRight` and `onPressArrowLeft` as defaults for `CalendarList`.